packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -1854,6 +1855,27 @@ describe('TransactionController', () => {
   });
 
   describe('stopTransaction', () => {
+    it('should avoid creating cancel transaction if transaction is confirmed', async () => {
+      const controller = newController();
+
+      controller.state.transactions.push({
+        id: '2',
+        chainId: toHex(5),
+        status: TransactionStatus.confirmed,
+        type: TransactionType.simpleSend,
+        time: 123456789,
+        txParams: {
+          from: ACCOUNT_MOCK,
+        },
+      });
+
+      await controller.stopTransaction('2');
+      expect(
+        pendingTransactionTrackerMock.checkTransactions,
+      ).toHaveBeenCalledTimes(1);
```

Comment:
We are not simulating `update` (thought could be overkill to implement it just for this test) but rather checking if `checkTransactions` is called.